src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java

```diff
@@ -0,0 +1,158 @@
+ * of an object general, but at the same time want to access it as one of two
+ * disconnected interfaces.
+ */
+public class SmartMotorController implements Sendable, SpeedController {
+
+    /** The wrapped motor controller, as a sendable object. */
+    private final Sendable sendable;
+    /** The wrapped motor controller. */
+    private final SpeedController wrapped;
+    /** An encoder, if one is attached and supplied. */
+    private final IEncoder encoder;
+    /** All the modifiers to apply to the speed controller. */
+    final private List<Modifier> modifiers;
+
+    /** Construct with mocks for everything */
+    public SmartMotorController() {
+        this(new MockSpeedController(), new MockEncoder());
+    }
+
+    /**
+     * Wrap a motor controller.
+     * 
+     * @param <T>       The base type to wrap
+     * @param wrapped   The wrapped motor controller.
+     * @param modifiers Any output modifiers.
+     */
+    public <T extends Sendable & SpeedController> SmartMotorController(final T wrapped, final Modifier... modifiers) {
+        this(wrapped, new MockEncoder(), modifiers);
+    }
+
+    /**
+     * Wrap a motor controller with an encoder.
+     * 
+     * @param <T>       The base type to wrap
+     * @param wrapped   The wrapped motor controller.
+     * @param encoder   The encoder attached to the motor controller.
+     * @param modifiers Any output modifiers.
+     */
+    public <T extends Sendable & SpeedController> SmartMotorController(final T wrapped, final IEncoder encoder,
+            final Modifier... modifiers) {
+        sendable = wrapped;
+        this.wrapped = wrapped;
+        this.encoder = encoder;
+        this.modifiers = new ArrayList<>(Arrays.asList(modifiers));
+    }
+
+    /**
+     * Get the encoder attached to the robot.
+     * 
+     * @return An encoder, or a mock if none is attached.
+     */
+    public IEncoder getEncoder() {
+        return encoder;
+    }
+
+    /**
+     * Add modifiers to the speed controller.
+     * 
+     * @param m  First modifier.
+     * @param ms Any extra modifiers (optional).
+     */
+    public void addModifiers(final Modifier m, final Modifier... ms) {
+        modifiers.add(m);
+        modifiers.addAll(Arrays.asList(ms));
+    }
+
+    /**
+     * Add all modifiers from a collection.
+     * 
+     * @param ms Collection of modifiers.
+     */
+    public void addAllModifiers(final Collection<? extends Modifier> ms) {
+        modifiers.addAll(ms);
+    }
+
+    /**
+     * Set the setpoint.
+     * 
+     * @param setPoint The new setpoint.
+     */
+    public void setSetpoint(final double setPoint) {
+        // Do nothing for this class
+    }
+
+    @Override
+    public void set(final double speed) {
+        wrapped.set(calculate(speed));
+    }
+
+    @Override
+    public double get() {
+        return wrapped.get();
+    }
+
+    @Override
+    public void setInverted(final boolean isInverted) {
```

Comment:
I wonder if it's worth adding a comment on this to remind us that we shouldn't be using this in the subsystem. Whether a motor is inverted is really a property of the robot itself.